### ISSUE TITLE
Support multi-tag searching

### ISSUE THREAD

**Issue by arweave-kyle** [ArweaveTeam/arweave-js#2]
Currently only a simple key/value search is possible, but we should support more complex queries with multiple tags.

**Current interface**

https://github.com/ArweaveTeam/arweave-js/blob/master/src/common/arweave/transactions.ts#L71
```typescript
search(tagName: string, tagValue: string): Promise<string[]>;
```
This method wraps the key and value into a single aqrl expression like this
```json
{
  "op": "equals", 
  "expr1": "Content-Type",
  "expr2": "text/html"
}
```
**New requirement**

An example of a more complex query that needs to be supported would look like this
```json
{
    "op": "and", 
    "expr1": {
	    "op": "or", 
	    "expr1": {
    		"op": "equals", 
    		"expr1": "Content-Type",
    		"expr2": "text/html"
	    },
	    "expr2": {
    		"op": "equals", 
    		"expr1": "Content-Type",
    		"expr2": "application/bdoc"
	    }
	},
    "expr2": {
	    "op": "equals", 
	    "expr1": "User-Agent",
	    "expr2": "ArweaveDeploy/1.0.5"
	}
}
```
Expressions can be nested and use combinations of `and`,`or`, and `equals` operators. IIRC there isn't a limit to nesting depth but I'll let @samcamwilliams confirm or clarify if there's a practical limit to consider.

**Comment by craze3** [comment]
Looks like this has been implemented already: https://github.com/ArweaveTeam/arweave-js#arql

We can close this issue @arweave-kyle :)

**Comment by arweave-kyle** [comment]
Ah yeah @craze3 good spot, thanks!

Fixed in cb6f49a.
